src/NLog/Layouts/Typed/Layout.cs

```diff
@@ -0,0 +1,315 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using JetBrains.Annotations;
+using NLog.Common;
+using NLog.Config;
+using NLog.Internal;
+
+namespace NLog.Layouts
+{
+    /// <summary>
+    /// Layout with a simple value (e.g. int) or a layout which results in a simple value (e.g. ${counter})
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class Layout<T> : Layout, IRenderable<T>
+    {
+        private static Type _type;
+        private static readonly string _typeNamed;
+        private readonly Layout _layout;
+        private readonly T _value;
+        private readonly bool _fixedValue;
+
+        /// <inheritdoc />
+        public Layout(T value)
+        {
+            _value = value;
+            _fixedValue = true;
+            _layout = null;
+        }
+
+        /// <inheritdoc />
+        public Layout(Layout layout)
+        {
+            _fixedValue = TryGetFixedValue(layout, out _value);
+            _layout = layout;
+        }
+
+        static Layout()
+        {
+            var type = typeof(T);
+            if (IsNullable(type))
+            {
+                var arg = type.GetGenericArguments()[0];
+
+#if !NETSTANDARD1_3 //todo fix
+
+                _type = arg;
+
+#endif
+            }
+            else
+            {
+                _type = type;
+            }
+
+            _typeNamed = type.Name;
+        }
+
+        private static bool IsNullable(Type propertyType)
+        {
+            return propertyType.IsGenericType() && propertyType.GetGenericTypeDefinition() == typeof(Nullable<>);
+        }
+
+        /// <summary>
+        /// Is fixed value?
+        /// </summary>
+        public bool IsFixed => _fixedValue;
+
+        #region Overrides of TypedLayout<T>
+
+        private static string ValueToString(T value, CultureInfo cultureInfo)
+        {
+            if (value is IConvertible convertible)
+            {
+                return convertible.ToString(cultureInfo);
+            }
+
+            return value?.ToString();
+        }
+
+        private static bool TryParse(string text, out T value)
+        {
+            return TryConvertTo(text, out value);
+        }
+
+        private static bool TryConvertTo(object raw, out T value)
+        {
+            if (_type == null || raw == null)
+            {
+                value = default(T);
+                return false;
+            }
+
+            // We don't use DI here because of this will be called before DI setup and performance reasons
+            var cultureInfo = CultureInfo.CurrentCulture;
+            try
+            {
+                var convertedValue = ValueConverter.Instance.Convert(raw, _type, null, cultureInfo);
```

Comment:
This doesn't look dependency injection friendly. Maybe fix it together with getting `Layout.FromString` working. Maybe extension method on LogFactory / ServiceRespository.

Reply:
This is indeed not DI friendly.

But altough I really like DI, im not sure if we need this here. Also this means we need to reference the LoggingConfiguration or LogFactory? We don't have those here (now).

Reply:
ow there is already a LoggingConfiguration there, missed that

Reply:
Fixed. What needs to be fixed in Layout.FromString?

Reply:
I was thinking that SimpleLayout should delay the parsing until being initialized with a LoggingConfiguration (during Target-initialization). Same with this Layout. Should not be doing any parsing until having been initialized.